File: USACO/AlphaStar/Silver Advanced/Problem Set 3/maxcross.py
N, K, B = map(int, input().split(" "))
signals = [1]*N
for _ in range(B):
    signals[int(input())-1] = 0

print(signals)
sums = [signals[0]]
for i in range(1, N):
    sums.append(sums[i-1] + signals[i])

print(sums)

fin = N-B
for m in range(1, N-B+1):
    for i in range(N-K):
        e = sums[i+K]-(0 if i == 0 else sums[i-1])
        if e + m >= K:
            print(f"{m} worked: {i}-{i+K}, {e} + {m} = {e+m} >= {K}")
            fin = m+1; break
    else:
        print(m, "didn't work")
        continue
    break

print(fin)

File: USACO/AlphaStar/Silver Advanced/Problem Set 3/maxcross.java
import java.util.Scanner;
public class maxcross {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int N = input.nextInt(), K = input.nextInt(), B = input.nextInt();
        int[] signals = new int[N+1];
        for (int i = 1; i < N+1; i++) signals[i] = 1;
        for (int i = 0; i < B; i++) signals[input.nextInt()] = 0;
        input.close();

        int[] sums = new int[N+1];
        for (int i = 1; i < N+1; i++) sums[i] = sums[i-1] + signals[i];

        int finK = N;
        outer: for (int k = 1; k < N; k++) {
            for (int i = 0; i < N+1-K; i++) {
                if (sums[i+K]-sums[i]+k >= K) {finK = k; break outer;}
            }
        }

        System.out.println(finK);
    }
}